### PR TITLE
Evolution of ABIS services technical specifications to support vendor-specific template format, quality format, and scoring information

### PR DESCRIPTION
**Pull request by olivier-heurtier-idemia**

### CODE REVIEW

File: src/doc/yaml/abis.yaml [hunk 1]
```diff
@@ -2101,21 +2101,16 @@ components:
...
-        - ISO_19794_2
-        - ISO_19794_2_NS
-        - ISO_19794_2_CS
-        - ISO_19794_2_2011
-        - ANSI_378_2009
-        - ANSI_378
-      description: Format of the template
+      description: |
+        Format of the template.
+        One of ISO_19794_2, ISO_19794_2_NS, ISO_19794_2_CS, ISO_19794_2_2011, ANSI_378_2009 or ANSI_378.
```
Comment (petr-sturc-thales) [inline src/doc/yaml/abis.yaml:2106]: Personally I prefer the enum approach. I would just add CUSTOM to the enum and maybe define optional string member subtype in  case implementer wants to precise which template/version it is.

File: src/doc/yaml/abis.yaml [hunk 2]
```diff
@@ -2101,21 +2101,16 @@ components:
...
       type: string
-      enum:
-        - ISO_19794
-        - NFIQ
-        - NFIQ2
-      description: Format of the quality
+      description: |
+        Format of the quality.
+        One of ISO_19794, NFIQ, or NFIQ2.
+        Can be extended to include additional proprietary quality format
```
Comment (petr-sturc-thales) [inline src/doc/yaml/abis.yaml:2113]: As per Template format. Enum seems safer to me, just with addition of CUSTOM as enum value. Just my opinion

Comment (srinivasjag) [reply src/doc/yaml/abis.yaml:2113]: With CUSTOM as the value, we will need to provide the actual value also. So where/how do we do that? A free form text would be better I feel.

File: src/doc/yaml/abis.yaml (hunk 1 above)
Comment (olivier-heurtier-idemia) [reply src/doc/yaml/abis.yaml:2106]: Hi @petr-sturc-thales ,
I understand your point and it is indeed a very valid proposition. I tend also to prefer enum when possible. Yet I propose to keep a string and not an enum for 2 reasons:
- First, the type is used to define a query parameter of the readTemplate operation. Adding a subtype would add a query parameter to this service with very little benefit for the consumer of the service
- Second, the type is used in the definition of type BiometricComputedData. This type is used only in responses, i.e. it is only built by the server and never by the client. I expect the server to apply appropriate coding rules, even enum if authorized by the programming language used, to make sure the returned value is correct.

Let me know what you think.
Same answer applies to your comment about QualityFormat.

**Pull request merged**